taichi/ir/type.h

```diff
@@ -166,12 +166,33 @@ class VectorType : public Type {
 
 class CustomIntType : public Type {
  public:
-  CustomIntType(int num_bits, bool is_signed)
-      : num_bits_(num_bits), is_signed_(is_signed) {
+  CustomIntType(int num_bits, bool is_signed);
+
+  CustomIntType(int compute_type_bits, int numBits, bool isSigned);
+
+  CustomIntType(int compute_type_bits,
+                Type *physical_type,
+                int num_bits,
+                bool is_signed);
```

Comment:
A single ctor `CustomIntType(int num_bits, bool is_signed, Type *physical_type_=nullptr, Type *compute_type_=nullptr)` may be cleaner?